File: LeetCode/Java/Easy/MergeSortedArray.java
package Easy;

public class MergeSortedArray {

	public static void main(String[] args){
		
	}
	
	public static void merge(int[] nums1, int m, int[] nums2, int n){
		int len = nums1.length-1;
		while (m > 0 && n > 0){
			if (nums1[m-1] >= nums2[n-1]){
				nums1[len] = nums1[m-1];
				m--;
			}
			else{
				nums1[len] = nums2[n-1];
				n--;
			}
			len--;
		}
		while (n > 0){
			nums1[len] = nums2[n-1];
			len--;
			n--;
		}
	}
}

File: LeetCode/Python/Easy/MergeSortedArray.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        m -= 1
        n -= 1

        insertionPoint = len(nums1) - 1
        while m >= 0 or n >= 0:
            if m < 0 or n < 0:
                nums1[insertionPoint] = nums1[m] if n < 0 else nums2[n]
                m -= 1
                n -= 1
            else:
                if nums1[m] > nums2[n]:
                    nums1[insertionPoint] = nums1[m]
                    m -= 1
                else:
                    nums1[insertionPoint] = nums2[n]
                    n -= 1
            insertionPoint -= 1
        return
